src/Stan/Config.hs

```diff
@@ -23,10 +42,27 @@ import Stan.Observation (Observation (..))
 import Stan.Severity (Severity (..))
 
 
-data Config = Config
-    { configChecks :: ![Check]
+data ConfigP (p :: Phase Text) = ConfigP
+    { configChecks :: !(p ::- [Check])
```

Comment:
What do you think about having the list of `Trial Check` in here? In addition to existing `Trial`s on top. So we can combine history for each `Check` individually, and in the final report we can write something like:

```
Check ... comes from TOML
Check ... comes from CLI
Check ... comes from TOML
```

Since we have `Traversable` instance for `Trial`, we can convert `[Trial e a]` to `Trial e [a]` and combine all nested histories into single top-level history.

Reply:
Turned out, I haven't thought about all details of my idea, and this might not be that helpful for users... As discussed, there are other ways to tell users in the report about checks and analysed inspections.